test/coordinator-test.js

```diff
@@ -0,0 +1,29 @@
+import { assert } from 'chai';
+import os from 'os';
+import sinon from 'sinon';
+import { getDefaultCPUs, getWorkerCount } from '../lib/coordinator';
+
+describe('coordinator', () => {
+  it('default method returns correct number of cpus', () => {
+    assert.equal(getDefaultCPUs(5), 4, 'getDefaultCPUs returns n - 1 CPUs');
+    assert.equal(getDefaultCPUs(), 1, 'getDefaultCPUs falls back to 1 CPU');
```

Comment:
(fwiw I only exported it for the sake of testing. i guess we could scope it inside of `getWorkerCount` to avoid this complexity, but i'm not a fan of this approach. lmk if you'd prefer this.)